mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AttachmentMapperTest.java

```diff
@@ -150,172 +146,84 @@ void getOwnerMessageIdsShouldReturnEmptyWhenNone() throws Exception {
-        //Then
-        assertThat(messageIds).containsOnly(messageId1, messageId2);
+        assertThat(attachmentMapper.getRelatedMessageIds(attachmentId)).containsOnly(messageId);
     }
 
     @Test
     void getOwnerMessageIdsShouldReturnOnlyMatchingMessageId() throws Exception {
-        //Given
-        Attachment attachment = Attachment.builder()
-                .bytes("payload".getBytes(StandardCharsets.UTF_8))
-                .type("content")
-                .build();
-        Attachment otherAttachment = Attachment.builder()
-                .bytes("something different".getBytes(StandardCharsets.UTF_8))
-                .type("content")
-                .build();
-        AttachmentId attachmentId = attachment.getAttachmentId();
         MessageId messageId1 = generateMessageId();
-        MessageId messageId2 = generateMessageId();
-        attachmentMapper.storeAttachmentsForMessage(ImmutableList.of(attachment), messageId1);
-        attachmentMapper.storeAttachmentsForMessage(ImmutableList.of(otherAttachment), messageId2);
-        
-        //When
-        Collection<MessageId> messageIds = attachmentMapper.getRelatedMessageIds(attachmentId);
-        //Then
-        assertThat(messageIds).containsOnly(messageId1);
-    }
-
-    @Test
-    void getOwnerMessageIdsShouldReturnOnlyOneMessageIdWhenStoredTwice() throws Exception {
-        //Given
-        Attachment attachment = Attachment.builder()
-                .bytes("payload".getBytes(StandardCharsets.UTF_8))
-                .type("content")
-                .build();
-        AttachmentId attachmentId = attachment.getAttachmentId();
-        MessageId messageId = generateMessageId();
-        attachmentMapper.storeAttachmentsForMessage(ImmutableList.of(attachment), messageId);
-        attachmentMapper.storeAttachmentsForMessage(ImmutableList.of(attachment), messageId);
-        
-        //When
-        Collection<MessageId> messageIds = attachmentMapper.getRelatedMessageIds(attachmentId);
-        //Then
-        assertThat(messageIds).containsOnly(messageId);
-    }
+        AttachmentId attachmentId1 = attachmentMapper.storeAttachmentsForMessage(ImmutableList.of(ParsedAttachment.builder()
+            .contentType("content")
+            .content(new ByteArrayInputStream("".getBytes(StandardCharsets.UTF_8)))
+            .noName()
+            .noCid()
+            .inline(false)), messageId1)
+            .get(0).getAttachmentId();
+        attachmentMapper.storeAttachmentsForMessage(ImmutableList.of(ParsedAttachment.builder()
+            .contentType("content")
+            .content(new ByteArrayInputStream("".getBytes(StandardCharsets.UTF_8)))
+            .noName()
+            .noCid()
+            .inline(false)), generateMessageId())
+            .get(0).getAttachmentId();
 
-    @Test
-    void getOwnerMessageIdsShouldReturnMessageIdForTwoAttachmentsWhenBothStoredAtTheSameTime() throws Exception {
```

Comment:
and what about this one?

Reply:
> getOwnerMessageIdsShouldReturnMessageIdForTwoAttachmentsWhenBothStoredAtTheSameTime

The attachmentIds being distinct, it cannot happen anymore.

"Protected by design" ;-)